ui/packages/catalog-ui-search/src/main/webapp/react-component/text-field/text-field.js

```diff
@@ -12,38 +12,71 @@
  * <http://www.gnu.org/licenses/lgpl.html>.
  *
  **/
-const React = require('react')
+import React from 'react'
+import PropTypes from 'prop-types'
 
-const Group = require('../group')
+import styled from '../styles/styled-components'
+import { readableColor, rgba } from 'polished'
 
-const CustomElements = require('../../js/CustomElements.js')
-const Component = CustomElements.registerReact('text-field')
+import Group from '../group'
+
+const foreground = props => {
```

Comment:
This function is repeated in each component, and the background one only differs in the `alpha` values. Can it be pulled out into a common module?